src/main/java/eu/dissco/core/digitalspecimenprocessor/service/DigitalMediaService.java

```diff
@@ -0,0 +1,136 @@
+package eu.dissco.core.digitalspecimenprocessor.service;
+
+import static eu.dissco.core.digitalspecimenprocessor.domain.EntityRelationshipType.HAS_MEDIA;
+
+import eu.dissco.core.digitalspecimenprocessor.domain.media.DigitalMediaEventWithoutDOI;
+import eu.dissco.core.digitalspecimenprocessor.domain.media.DigitalMediaKey;
+import eu.dissco.core.digitalspecimenprocessor.domain.media.DigitalMediaProcessResult;
+import eu.dissco.core.digitalspecimenprocessor.domain.specimen.DigitalSpecimenRecord;
+import eu.dissco.core.digitalspecimenprocessor.domain.specimen.UpdatedDigitalSpecimenRecord;
+import eu.dissco.core.digitalspecimenprocessor.repository.DigitalMediaRepository;
+import eu.dissco.core.digitalspecimenprocessor.schema.EntityRelationship;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class DigitalMediaService {
+
+  private final DigitalMediaRepository mediaRepository;
+
+  public Map<String, DigitalMediaProcessResult> getExistingDigitalMedia(
+      Map<String, DigitalSpecimenRecord> currentSpecimens,
+      List<DigitalMediaEventWithoutDOI> mediaEvents) {
+    var mediaProcessResults = new HashMap<String, DigitalMediaProcessResult>();
+    var mediaMap = new HashMap<String, List<DigitalMediaEventWithoutDOI>>();
+    mediaEvents.forEach(mediaEvent ->
+        mediaMap.computeIfAbsent(mediaEvent.digitalMediaObjectWithoutDoi().physicalSpecimenID(),
```

Comment:
Just to check. So if a physicalSpecimenID in the map has a null value, you add a new list with the event? I don't fully get this, the map it empty at the start so it will always hit the Absent part right? Why not use the List.of() if you always add a single event?

Reply:
Ah yes, thanks for the clarification